docgen/parser/parser.go

```diff
@@ -155,14 +155,63 @@ func parseValues(path string, comments map[string]*Comment, values []*google_pro
 	return res
 }
 
+// extractMapFields generates an array of fields that correspond to map type fields for messages, which are stored in
+// the NestedType field of the DescriptorProto struct
+func extractMapFields(message *google_protobuf.DescriptorProto, path string, comments map[string]*Comment) []*Field {
+	fields := make([]*Field, len(message.NestedType))
+	for i, n := range message.GetNestedType() {
+		// a nested type of map will always have two fields for key and value
+		if len(n.GetField()) != 2 {
+			continue
+		}
+		nestedTypeName := strings.TrimSuffix(n.GetName(), "Entry")
+		// find position and fieldDescriptorProto of the map type from the message fields
+		pos := -1
+		var fieldDescriptorProto *google_protobuf.FieldDescriptorProto
+		for i, f := range message.GetField() {
+			if f.GetName() == nestedTypeName {
+				pos = i
+				fieldDescriptorProto = f
+				break
+			}
+		}
+		if pos == -1 || fieldDescriptorProto == nil {
+			continue
+		}
+
+		// get key type. key can only be string or int32
```

Comment:
if this is the case, again, please do a `panic` otherwise - with the message here. otherwise we could silently change some other type for `string`, potentially.

Reply:
Or just use `getType`?

Reply:
I can't use getType because it returns a FieldDescriptorProto_Type type const.